File: commons/javafx/src/main/java/com/cedarsoft/commons/javafx/ColorUtil.java
package com.cedarsoft.commons.javafx;

import javax.annotation.Nonnull;

import javafx.scene.paint.Color;

/**
 * A utility class around {@link Color}.
 */
public class ColorUtil {

  private ColorUtil() {
    // a utility class
  }

  @Nonnull
  public static String toRGB(@Nonnull Color color) {
    int r = (int) (color.getRed() * 255);
    int g = (int) (color.getGreen() * 255);
    int b = (int) (color.getBlue() * 255);
    return "rgb(" + r + ", " + g + ", " + b + ")";
  }

  @Nonnull
  public static String toRGBA(@Nonnull Color color) {
    int r = (int) (color.getRed() * 255);
    int g = (int) (color.getGreen() * 255);
    int b = (int) (color.getBlue() * 255);
    return "rgb(" + r + ", " + g + ", " + b + ", " + color.getOpacity() + ")";
  }

}

File: commons/javafx/src/main/kotlin/it/neckar/open/javafx/ColorUtil.kt
package it.neckar.open.javafx

import javafx.scene.paint.Color
import javax.annotation.Nonnull

/**
 * A utility class around [Color].
 */
object ColorUtil {
  @JvmStatic
  fun toRGB(@Nonnull color: Color): String {
    val r = (color.red * 255).toInt()
    val g = (color.green * 255).toInt()
    val b = (color.blue * 255).toInt()
    return "rgb($r, $g, $b)"
  }

  @JvmStatic
  fun toRGBA(@Nonnull color: Color): String {
    val r = (color.red * 255).toInt()
    val g = (color.green * 255).toInt()
    val b = (color.blue * 255).toInt()
    return "rgb(" + r + ", " + g + ", " + b + ", " + color.opacity + ")"
  }
}
